addons/binding/org.openhab.binding.smappee/src/main/java/org/openhab/binding/smappee/handler/SmappeeHandler.java

```diff
@@ -0,0 +1,142 @@
+                updateStatus(ThingStatus.OFFLINE);
+            }
+        } else {
+            logger.debug("Command {} is not supported for channel: {}", command, channelUID.getId());
+        }
+    }
+
+    @Override
+    public void newState(SmappeeDeviceReading readings) {
+        updateState(CHANNEL_CONSUMPTION, new DecimalType(readings.getLatestConsumption()));
+        updateState(CHANNEL_SOLAR, new DecimalType(readings.getLatestSolar()));
+        updateState(CHANNEL_ALWAYSON, new DecimalType(readings.getLatestAlwaysOn()));
+    }
+
+    @Override
+    public void invalidConfig() {
+        updateStatus(ThingStatus.OFFLINE);
+    }
+
+    @Override
+    public void initialize() {
+        Configuration conf = this.getConfig();
+
+        String param_client_id = String.valueOf(conf.get(PARAMETER_CLIENT_ID));
+        String param_client_secret = String.valueOf(conf.get(PARAMETER_CLIENT_SECRET));
+        String param_username = String.valueOf(conf.get(PARAMETER_USERNAME));
+        String param_password = String.valueOf(conf.get(PARAMETER_PASSWORD));
+        String param_serviceLocationName = String.valueOf(conf.get(PARAMETER_SERVICE_LOCATION_NAME));
+        String param_polltime = String.valueOf(conf.get(PARAMETER_POLLTIME));
+
+        if (param_client_id.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Check configuration, Client Id must be provided");
+            return;
+        }
+        if (param_client_secret.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Check configuration, Client secret must be provided");
+            return;
+        }
+        if (param_username.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Check configuration, Username must be provided");
+            return;
+        }
+        if (param_password.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Check configuration, Password must be provided");
+            return;
+        }
+        if (param_serviceLocationName.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Check configuration, Service location name must be provided");
+            return;
+        }
+        if (param_polltime.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Check configuration, polling time must be provided");
+            return;
+        }
+
+        int polltime = Integer.parseInt(param_polltime) * 60000;
+
+        logger.debug("Initialize Network handler.");
+        smappeeService = new SmappeeService(param_client_id, param_client_secret, param_username, param_password,
+                param_serviceLocationName, polltime);

```

Comment:
I see both `polltime` and `refreshInterval` used for what seems to be the same thing.

Reply:
changed RefreshInterval to pollTime (it's the same)